clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -652,7 +652,8 @@ public Node leastLoadedNode(long now) {
             int idx = (offset + i) % nodes.size();
             Node node = nodes.get(idx);
             int currInflight = this.inFlightRequests.count(node.idString());
-            if (currInflight == 0 && isReady(node, now)) {
+            if (currInflight == 0 &&
+                    (this.metadataUpdater.isUpdateDue(now) ? canSendRequest(node.idString(), now) : isReady(node, now))) {
```

Comment:
I think we may be able to simplify this and just use `canSendRequest`. If a metadata update is due (which is a rare case), we will still be prevented from sending. It doesn't seem like we get any benefit from having `leastLoadedNode` choose a different node which may have in-flight requests.